parser.rb

```diff
@@ -0,0 +1,41 @@
+
+class Parser
+  def initialize(filename)
+    file = File.open(filename, "r")
+    @text = ''
+    file.each_line do |txt|
```

Comment:
Right here you're iterating over each line of the file and appending them onto @text one at a time, but there's a method you could use to just read the whole file at once and get the same result. Check out File#read